com.unity.ml-agents/Runtime/Inference/TensorProxy.cs

```diff
@@ -69,13 +69,21 @@ public long Channels
 
         ~TensorProxy()
         {
-            data?.Dispose();
+            Dispose();
+        }
+
+        void Dispose()
+        {
+            if (data.dataOnBackend.backendType != BackendType.CPU)
+            {
+                data?.Dispose();
```

Comment:
Why is it not needed to dispose when using CPU?

Reply:
There's no need to copy to the CPU mem. It's already there and accessible. This really only applies, at least for us, when using GPU backends.